src/Common/Commands.Common/MetricHelper.cs

```diff
@@ -226,6 +226,16 @@ private void LogExceptionEvent(AzurePSQoSEvent qos)
                 eventProperties.Add("Message", "Message removed due to PII.");
                 eventProperties.Add("StackTrace", qos.Exception.StackTrace);
                 eventProperties.Add("ExceptionType", qos.Exception.GetType().ToString());
+                Exception innerEx = qos.Exception.InnerException;
+                int exceptionCount = 0;
+                //keep goin till we get to the last inner exception
+                while (innerEx != null)
+                {
+                    //Increment the inner exception count so that we can tell which is the outermost
+                    //and which the innermost
+                    eventProperties.Add("InnerExceptionType-"+exceptionCount++, innerEx.GetType().ToString());
```

Comment:
@cormacpayne I had initially missed this piece in the linked issue :"InnerExceptionType (shouls contain a list of inner exception types)"

Hence I am adding all the inner exceptions. I can create a comma separated list of exceptions and add it to the dictionary, but I thought it might be better to return each one as a separate item in the dictionary.